### PR TITLE
Develop Table Uploads: Replace

### PR DESCRIPTION
**Pull request by GiaJordan**
When uploading a manifest to synapse and storing as a Table, if the table exists and has metadata from another manifest already, the new manifest's metadata will be appended to the existing table, and some of the existing metadata will be replaced with dashes "-".  This was half of the requests from multiple issues #834 #911.
The current intended behavior for schematic in this scenario is to replace the existing data entirely, and only store the metadata from the new manifest. The [`replace_table`](https://github.com/Sage-Bionetworks/schematic_db/blob/develop/synapse/synapse.py#L241) method from schematic_db(@andrewelamb) has been adapted and introduced to the schematic synapse store's `make_table` method.

Further PRs will address feature requests from the two earlier mentioned issues, as well as splitting table behavior for censored and uncensored manifests.

### CODE REVIEW

**Comment by lopierra** [comment]
@GiaJordan I'm not sure how to test this - is there something specific @brynnz22 and I should do?

**Comment by GiaJordan** [comment]
@lopierra I'm currently releasing a containerized version of this branch on docker that you can use schematic in to test that the manifest table uploads work as expected! You, @brynnz22, and I can also work to test it together if need be!

**Comment by GiaJordan** [comment]
@lopierra and @brynnz22 
The [container](https://hub.docker.com/layers/sagebionetworks/schematic/commit-6da9c93/images/sha256-7790ee4f5ebd6fce244c5be456effade286fc41fdae3ab8f0a2087468b92eceb?context=explore) is now up for you to download and use to test!

**Comment by brynnz22** [comment]
@GiaJordan can we test this together? Perhaps on Thursday or Friday? I think both @lopierra and I are not super stars with Docker and it would be helpful to go through this.

File: schematic/store/synapse.py [hunk 1]
```diff
@@ -1317,20 +1350,73 @@ def getDatasetAnnotationsBatch(
...
+            # locate its position in the old and new table.
+            if manipulation == 'update':
+                table_to_load = update_df(existing_table, table_to_load, update_col)
+            
+            elif manipulation == 'replace':
+                # remove rows
+                self.syn.delete(existing_results)
+
+                # wait for row deletion to finish on synapse before getting empty table
+                sleep(1)
```
Comment (milen-sage) [inline schematic/store/synapse.py:1401]: @GiaJordan @andrewelamb do you think the sleep interval is always enough? It seems a bit risky?

**Review by milen-sage — COMMENTED** [review]
Please see comment below. Otherwise it looks good.

File: schematic/store/synapse.py (hunk 1 above)
Comment (GiaJordan) [reply schematic/store/synapse.py:1401]: In testing I had decreased it from 30 seconds in intervals to arrive at the 1 second. But after second thought we could increase it to be on the safe side for larger tables that might take more time?
@milen-sage @andrewelamb

**Review by andrewelamb — APPROVED** [review]

File: schematic/store/synapse.py (hunk 1 above)
Comment (GiaJordan) [reply schematic/store/synapse.py]: @milen-sage  increased to 5sec

**Comment by GiaJordan** [comment]
@brynnz22 and @lopierra thanks for your patience as we sorted out the short term future of this branch. It's not been merged to `develop` and will also be included in the next release, so you'll also be able to use this feature by pulling the current latest `develop` or by installing the newest schematic version after #934 is merged. 
If you only wanted assistance with using docker then feel free to go ahead and use the `develop` branch or install the upcoming release to use this feature; otherwise we can still find another time to meet about the actual use of the table feature since the working session time doesn't work for all of us.

@/afwillia and I will also be working this sprint to put together a set of instructions for installing and using docker to test features: #924

**Comment by brynnz22** [comment]
Sounds good. Thanks so much @GiaJordan!

**Pull request merged**